samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -34,32 +34,111 @@
 #include "sms.h"
 #endif
 
-#if defined(CONFIG_MODEM_INFO)
 #include <modem/at_cmd_parser.h>
 #include <modem/at_params.h>
-#endif
 
 static bool link_subscribe_for_rsrp;
 
 extern const struct shell *shell_global;
 
 extern bool uart_disable_during_sleep_requested;
 
-#if defined(CONFIG_MODEM_INFO)
+struct pdn_activation_status_info {
+	bool activated;
+	uint8_t cid;
+};
+
 /* System work queue for getting the modem info that ain't in lte connection ind: */
-static struct k_work modem_info_work;
+static struct k_work registered_work;
 
 /* Work queue for signal info: */
 static struct k_work modem_info_signal_work;
 #define LINK_RSRP_VALUE_NOT_KNOWN -999
 static int32_t modem_rsrp = LINK_RSRP_VALUE_NOT_KNOWN;
 
+/* Work queue for continuous neighbor cell measurements: */
+static struct k_work continuous_ncellmeas_work;
+enum link_ncellmeas_modes ncellmeas_mode = LINK_NCELLMEAS_MODE_NONE;
+
+static void link_continuous_ncellmeas(struct k_work *work)
+{
+	if (ncellmeas_mode == LINK_NCELLMEAS_MODE_CONTINUOUS) {
+		link_ncellmeas_start(true, LINK_NCELLMEAS_MODE_CONTINUOUS);
+	}
+}
+
 /******************************************************************************/
 
-static void link_modem_info_work(struct k_work *unused)
+static void link_api_activate_mosh_contexts(
+	struct pdn_activation_status_info pdn_act_status_arr[], int size)
+{
+	int i, esm;
+
+	/* Check that all context created by mosh link connect are active and if not,
+	 * then activate:
```

Comment:
There are many of these lines that would fit into 100 character lines. I'll leave it to you whether to change them or not but think about it especially in future changes.

Reply:
Yes, did some fixes elsewhere but this particular one does not fit to 100.